components/scream/src/physics/spa/spa_functions_impl.hpp

```diff
@@ -90,195 +90,204 @@ void SPAFunctions<S,D>
 ::spa_main(
   const SPATimeState& time_state,
   const view_2d<const Spack>& p_tgt,
-  const SPAData&   data_beg,
-  const SPAData&   data_end,
-  const SPAOutput& data_out,
-  Int ncols_tgt,
-  Int nlevs_tgt,
-  Int nswbands,
-  Int nlwbands)
+  const view_2d<      Spack>& p_src,
+  const SPAInput&   data_beg,
+  const SPAInput&   data_end,
+  const SPAInput&   data_tmp,
+  const SPAOutput&  data_out)
+{
+  // Beg/End/Tmp month must have all sizes matching
+  EKAT_REQUIRE_MSG (
+      data_end.data.nswbands==data_beg.data.nswbands &&
+      data_end.data.nswbands==data_tmp.data.nswbands &&
+      data_end.data.nlwbands==data_beg.data.nlwbands &&
+      data_end.data.nlwbands==data_tmp.data.nlwbands,
+      "Error! SPAInput data structs must have the same number of SW/LW bands.\n");
+  EKAT_REQUIRE_MSG (
+      data_end.data.ncols==data_beg.data.ncols &&
+      data_end.data.ncols==data_tmp.data.ncols &&
+      data_end.data.nlevs==data_beg.data.nlevs &&
+      data_end.data.nlevs==data_tmp.data.nlevs,
+      "Error! SPAInput data structs must have the same number of columns/levels.\n");
+
+  // Output must have same number of bands
+  EKAT_REQUIRE_MSG (
+      data_end.data.nswbands==data_out.nswbands &&
+      data_end.data.nlwbands==data_out.nlwbands,
+      "Error! SPAInput and SPAOutput data structs must have the same number of SW/LW bands.\n");
+
+  // While horiz interp is not supported, we also require matching ncols
+  EKAT_REQUIRE_MSG (
+      data_end.data.ncols==data_out.ncols,
+      "Error! While horizontal interpolation is not supported in spa_main,\n"
+      "       SPAInput and SPAOutput data structs must have the same number columns.\n");
+
+  // Step 1. Perform horizontal interpolation (todo)
```

Comment:
See comment above.  Maybe "Step 1. Perform horizontal interpolation (already done in `update_spa_data`"